src/MigrationDeriver.php

```diff
@@ -175,12 +175,11 @@ protected function deriveMigrationName($mg_name, $target) {
    *   TRUE if it is the same; otherwise, FALSE.
    */
   protected function sameMigrationGroup(MigrationInterface $mig, $target) {
+    /** @var \Drupal\migrate\Plugin\MigrationInterface $loaded_target */
     $loaded_target = $this->migrationPluginManager->createInstance($target);
-    // XXX: General getters were deprecated and removed in:
-    // https://www.drupal.org/node/2873795. Given how migrate_plus injects
-    // the group need to get it without it.
-    $mg = $loaded_target->migration_group;
-    return $mg && $mg == $mig->migration_group;
+
+    $mg = $loaded_target->getPluginDefinition()['migration_group'] ?? FALSE;
+    return $mg && ($mg === ($mig->getPluginDefinition()['migration_group'] ?? FALSE));
```

Comment:
Instead of the janky dynamic references that seem magickal, we can reference from the plugin definition directly.